src/txn/txn.c

```diff
@@ -462,7 +462,8 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 
 	txn = &session->txn;
 	conn = S2C(session);
-	WT_ASSERT(session, !F_ISSET(txn, WT_TXN_ERROR));
+	if (txn->mod_count !=0)
+		WT_ASSERT(session, !F_ISSET(txn, WT_TXN_ERROR));
```

Comment:
These two lines should just be:

```c
WT_ASSERT(session, !F_ISSET(txn, WT_TXN_ERROR) || txn->mod_count == 0);
```

(with appropriate wrapping).

The reason for this is that the check inside the assertion doesn't run in a release build, whereas the "if" check does.